TFastModbus.cpp

```diff
@@ -0,0 +1,173 @@
+#include "TFastModbus.h"
+#include "Arduino.h"
+#include "CrcClass.h"
+#include "Debug.h"
+#include "DebugOutput.h"
+#include "Status.h"
+
+#define TIMEOUT_COUNTING_DELTA 0xA
+
+TFastModbus::TFastModbus(HardwareSerial* hardwareSerial, uint16_t timeoutMs)
+    : Serial(hardwareSerial),
+      TimeoutMs(timeoutMs)
+{}
+
+bool TFastModbus::OpenPort(size_t speed, uint32_t config, uint8_t rx, uint8_t tx)
+{
+    return (Serial->begin(speed, config, rx, tx) == ZunoErrorOk);
+}
+
+// Starts scan cycle
+bool TFastModbus::StartScan()
+{
+    // Fast modbus packet with start scan command
+    // 0xFD - address for scan, 0x60 command, 0x01 - start subcommand
+    uint8_t startData[] = {0xFD, 0x60, 0x01, 0x09, 0xF0};
+    bool result = Serial->write(startData, sizeof(startData)) == sizeof(startData);
+
+    if (result)
+        delay(Serial->countWaitingMs(4));
+    else {
+        DEBUG("*** ERROR Sending fast modbus scan start!\n");
+    }
```

Comment:
```suggestion
    if (!result) {
         DEBUG("*** ERROR Sending fast modbus scan start!\n");
         return false;
     }

     delay(Serial->countWaitingMs(4));
     return true;
```

И кстати, зачем тут delay?

Reply:
Потому что модбас требует чтобы между пакетами был промежуток времени на 3,5 символа. Соответственно, если отправка не удалась, то можно не ждать: все равно ответа не будет.